model/task_history.go

```diff
@@ -389,35 +391,20 @@ func (self *taskHistoryIterator) GetDistinctTestNames(ctx context.Context, env e
 			{"$unwind": fmt.Sprintf("$%v", testResultsKey)},
 			{"$group": bson.M{"_id": fmt.Sprintf("$%v.%v", testResultsKey, task.TestResultTestFileKey)}},
 		},
-		opts,
 	)
-	if err != nil {
-		return nil, errors.Wrapf(err, "error getting cursor")
-	}
-	if cursor == nil {
-		return nil, errors.New("nil cursor returned")
-	}
-	defer cursor.Close(ctx)
-
-	names := []string{}
-	for cursor.Next(ctx) {
-		select {
-		case <-ctx.Done():
-			return nil, errors.New("context cancelled")
-		default:
-			res := bson.M{}
-			if err := cursor.Decode(&res); err != nil {
-				return nil, errors.Wrapf(err, "error decoding result")
-			}
-			// remove quotes from value
-			val := strings.Replace(res["_id"].(string), "\"", "", -1)
-			names = append(names, val)
+	pipeline.MaxTime(time.Minute)
+
+	var names []string
+	res := bson.M{}
+	iter := pipeline.Iter()
+	for iter.Next(&res) {
+		names = append(names, res["_id"].(string))
+		if ctx.Err() != nil {
+			_ = iter.Close()
```

Comment:
Is this supposed to ignore the error? The typical way to deal with closing resources during an error case is to accumulate them in a grip catcher.

Reply:
that's an excellent idea.
done